### PR TITLE
Fix UI bugs for openstack integration

### PR DESCRIPTION
**Pull request by jcwest**
- unable to destroy server group
  - infrastructureCaches error in console
  - multiselect drop-downs not working

### CODE REVIEW

File: app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js [hunk 1]
```diff
@@ -31,22 +31,37 @@ module.exports = angular.module('spinnaker.openstack.cacheBackedMultiSelectField
...
+        //update the options whenever the associated cache gets updated
         scope.$watch('cache', updateOptions);
+
+        //The following watchers are needed to overcome the fact that ui-select requires the model to be a child
+        //property.  This code effectively removes that constraint for users of this directive.
+
+        //carry through any changes from the UI to the provided model
+        scope.$watch('state.selectedOptions', function() {
+            //make sure the update is coming from the UI and not from the parent
+            if( scope.model != scope.state.selectedOptions ) {
```
Comment (danielpeach) [inline app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js]: s/`!=`/`!==`

File: app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js [hunk 2]
```diff
@@ -31,22 +31,37 @@ module.exports = angular.module('spinnaker.openstack.cacheBackedMultiSelectField
...
+                scope.onChange(args);
+              }
+            }
+        });
+
+        //make sure any changes in the model are reflected in the UI
         scope.$watch('model', function(model) {
           scope.state.selectedOptions = model;
         });
+
```
Comment (danielpeach) [inline app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js]: Extraneous lines

File: app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js [hunk 3]
```diff
@@ -31,22 +31,37 @@ module.exports = angular.module('spinnaker.openstack.cacheBackedMultiSelectField
...
-              scope.onChange(args);
-            }
           }
         });
 
         function updateOptions() {
           scope.options = _.sortBy(scope.cache, 'name');
         }
 
+        //update the options whenever the associated cache gets updated
```
Comment (danielpeach) [inline app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js]: Please add a space (`// update ...`) here and below.

File: app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js (hunk 1 above)
Comment (danielpeach) [inline app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js]: Please reformat as `if (scope.model !== scope.state.selectedOptions) {` here and for the other if statements below.

**Comment by jcwest** [comment]
Thanks for the review @danielpeach! I updated the code per your feedback.

**Pull request merged**